# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/WebGPU.webidl
# Commit: a5be865d06b1
# Full Hash: a5be865d06b167e20cff16ad33e4f076bbe41757
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-07-17 09:30:54
# Regressor Bug: 1976960
# File Overlap Count: 1
# Description:
#   Bug 1976960 - feat(webgpu): allow `GPUTexture`s where `GPUTextureView`s are allowed r=webgpu-reviewers,webidl,smaug,nical
#   
#   Changed in spec. upstream by [gpuweb#5228](https://github.com/gpuweb/gpuweb/pull/5228), tested by CTS in [gpuweb/cts#4407](https://github.com/gpuweb/cts/pull/4407) (which is what caused the `FAIL`s removed here to be introduced with bug 1976874).
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D257054
# ==============================================================================

diff -r e98afaa9f97b -r a5be865d06b1 dom/webidl/WebGPU.webidl
--- a/dom/webidl/WebGPU.webidl	Thu Jul 17 04:08:37 2025 +0000
+++ b/dom/webidl/WebGPU.webidl	Thu Jul 17 04:08:38 2025 +0000
@@ -598,6 +598,7 @@
 };
 
 typedef (GPUSampler or
+         GPUTexture or
          GPUTextureView or
          GPUBufferBinding) GPUBindingResource;
 
@@ -1120,9 +1121,9 @@
 };
 
 dictionary GPURenderPassColorAttachment {
-    required GPUTextureView view;
+    required (GPUTexture or GPUTextureView) view;
     GPUIntegerCoordinate depthSlice;
-    GPUTextureView resolveTarget;
+    (GPUTexture or GPUTextureView) resolveTarget;
 
     GPUColor clearValue;
     required GPULoadOp loadOp;
@@ -1130,7 +1131,7 @@
 };
 
 dictionary GPURenderPassDepthStencilAttachment {
-    required GPUTextureView view;
+    required (GPUTexture or GPUTextureView) view;
 
     float depthClearValue;
     GPULoadOp depthLoadOp;
